lib/charms/kubernetes_charm_libraries/v0/volumes.py

```diff
@@ -0,0 +1,453 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm Library used to leverage the Volumes Kubernetes in charms."""
+import logging
+from dataclasses import dataclass
+from typing import Iterable
+
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.models.apps_v1 import StatefulSetSpec
+from lightkube.models.core_v1 import (
+    Container,
+    PodSpec,
+    PodTemplateSpec,
+    ResourceRequirements,
+    Volume,
+    VolumeMount,
+)
+from lightkube.resources.apps_v1 import StatefulSet
+from lightkube.resources.core_v1 import Pod
+from lightkube.types import PatchType
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class RequestedVolume:
+    """RequestedVolume."""
+
+    volume: Volume
+    volume_mount: VolumeMount
+
+
+class KubernetesRequestedVolumesError(Exception):
+    """KubernetesRequestedVolumesError."""
+
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(self.message)
+
+
+class KubernetesClient:
+    """Class containing all the Kubernetes specific calls."""
+
+    def __init__(self, namespace: str):
+        self.client = Client()
+        self.namespace = namespace
+
+    def pod_is_patched(
+        self,
+        pod_name: str,
+        *,
+        requested_volumes: Iterable[RequestedVolume],
+        container_name: str,
+    ) -> bool:
+        """Returns whether pod has the requisite requested volumes and resources (when required).
```

Comment:
A pod is considered patched when it contains the requested volumes.
In case of a HugePage volume request, it must contain proper resources (limits and requests).

Or something like that...